client/src/components/App/index.js

```diff
@@ -32,14 +32,18 @@ function App() {
           <Route
             exact
             path="/profile/parent/:id"
-            render={() => <ParentProfile />}
+            render={props => <ParentProfile {...props} />}
           />
           <Route
             exact
             path="/profile/teacher/:id"
             render={props => <TeacherProfile {...props} />}
           />
-          <Route exact path="/student/:id" render={() => <StudentProfile />} />
+          <Route
+            exact
+            path="/student/:id"
+            render={props => <StudentProfile {...props} />}
```

Comment:
unless I'm missing something using props here has no effect as they will be always empty (the `match`, `location` and `history` properties should be passed automatically). So should be enough to do `render={() => <StudentProfile />}`, which can be changed to `component={StudentProfile}`

Reply:
i use render for functioal component and properties shouldn't be passed automatically

Reply:
but understand that by using `props => <StudentProfile {...props} />` you are wrapping your functional component `StudentProfile` in another functional component that just creates the StudentProfile component essentially in the same way as if you used `component={StudentProfile}`. If you wanna use `render` for functional component (as it does have some performance benefits by skipping the lifecycle methods), use it like this `render={StudentProfile}`.
The properties are [passed automatically](https://stackoverflow.com/questions/52070532/render-props-react-route) by the router.